packages/cli/src/commands/ls.ts

```diff
@@ -19,7 +19,12 @@ export default class Ls extends DriverCommand<typeof Ls> {
   async run(): Promise<unknown> {
     const { flags } = await this.parse(Ls)
     const driver = await this.driver()
-    const machines = await asyncToArray(await commands.ls({ machineDriver: driver, log: this.logger }))
+    const machines = await asyncToArray(asyncMap(x => {
+      if ('publicIPAddress' in x) {
```

Comment:
can probably benefit from defining
```typescript
const isFullMachine = (o: PartialMachine): o is Machine => typeof (o as Machine).publicIPAddress === 'string'
```

then
```typescript
return { ...x, partial: !isFullMachine(x) }
```

and I think it can be used elsewhere as well.